### ISSUE TITLE
Responsive styles

### ISSUE THREAD

**Issue by lnpbk** [mike-engel/styled-typography#1]
Is there any solution for handling responsive styles?

I'm interested to know if it's possible to increase a heading's font size as screen size increases

**Comment by mike-engel** [comment]
@lnpbk The way to do that now would be to update your `theme`'s `fontSizes` array as the window size changes. If you have an idea for an API that handles mobile sizes and desktop sizes, I'd be interested to discuss it!

**Comment by lnpbk** [comment]
@mike-engel The only way I can think of handling it would be to have some sort of map between breakpoints and levels.

```jsx
<Heading
    displayLevels={{ break1: 1, break2: 2}}
    ....
/>
```

styled-typography would either have to know about the breakpoints, or have them passed into that object with this set-up though

**Comment by mike-engel** [comment]
Yeah, I think that's more trouble than it's worth. An API using `matchmedia`, might work, however. Something like so:

```js
{
  fontSizes: {
    "min-width: 320px": [...],
    "min-width: 720px": [...],
    "min-width: 1024px": [...],
    "screen": [...]
  }
}
```

The main problem there is that order is not guarunteed when going through an object, so I would have to be able to sort every valid `matchmedia` statement, which wouldn't be great. An array of arrays may be possible, however:

```js
{
  fontSizes: [
    ["min-width: 320px", [...]],
    ["min-width: 720px", [...]],
    ["min-width: 1024px", [...]],
    ["default", [...]]
  ],
}
```

With that approach, the last match would be the one selected. So if you used `max-width` instead, it would need to be flipped:
```js
{
  fontSizes: [
    ["max-width: 1024px", [...]],
    ["max-width: 720px", [...]],
    ["max-width: 320px", [...]],
    ["default", [...]]
  ],
}
```

How would you feel about something like that? Would that solve your problem? Then again, this might just be simpler to keep in user-space 😄

**Comment by jbmusso** [comment]
Hello, thanks for the great library!

I am ensure if you've had the time to think about this yet? I'm currently experimenting with https://github.com/artsy/fresnel (also tried https://github.com/contra/react-responsive), I'd be happy to help craft a solution for this.

Cheers

**Comment by mike-engel** [comment]
@jbmusso unfortunately I haven't had much time to think about this. If you would like to make a PR, I'd be happy to review it, however.
